src/main/java/org/folio/dew/batch/eholdings/GetEHoldingsWriter.java

```diff
@@ -47,9 +46,10 @@ public void write(List<? extends EHoldingsResourceDTO> list) throws Exception {
       jobExecution.getExecutionContext().getInt(CONTEXT_TOTAL_RESOURCES, 0) + resources.size());
 
     var resourceWithMaxNotes = list.stream()
+      .filter(p -> p.getNotes() != null)
       .max(Comparator.comparing(p -> p.getNotes().size()))
-      .orElseThrow(NoSuchElementException::new);
-    var noteCollectionSize = resourceWithMaxNotes.getNotes().size();
+      .orElse(null);
+    var noteCollectionSize = resourceWithMaxNotes == null ? 0 : resourceWithMaxNotes.getNotes().size();
```

Comment:
maybe it is better to use @Singular annotation like this:
![image](https://user-images.githubusercontent.com/13659775/194526209-beb31028-a9ef-4100-b8e6-a87c5122f240.png)